vignettes/Model.Rmd

```diff
@@ -10,90 +12,365 @@ vignette: >
+
+### Species distribution
+
+`malariasimulation` contains in built parameter sets for three mosquito species: *A. arabiensis* (`arab_params`), *A. funestus* (`fun_params`) and *A. gambiae* (`gamb_params`). Each parameter set contains feeding rates, foraging time etc. Full details can be found in the [vector_parameters.R](https://github.com/mrc-ide/malariasimulation/blob/master/R/vector_parameters.R) file.
+
+```{r}
+unlist(arab_params)
+```
+
+The proportion of each species can be set using the `set_species()` function, requiring a full parameter set and arguments of a list of species parameter sets and a vector of relative species proportions.
+
+```{r, fig.align = 'center', fig.height = 5, fig.width = 7}
+# Create list of default parameters
+params <- get_parameters()
+
+# Update parameter list with species distributions
+params_species <- set_species(parameters = params,
+                              species = list(arab_params, fun_params, gamb_params),
+                              proportions = c(0.1,0.3,0.6))
+
+# Run simulation
+species_simulation <- run_simulation(timesteps = 100, parameters = params_species)
+
+## Plot species distributions
+# Collect species column names
+arab_cols <- paste0(c("E","L","P","Sm","Pm", "Im", "NonExistent"),"_arab_count")
+fun_cols <- paste0(c("E","L","P","Sm","Pm", "Im", "NonExistent"),"_fun_count")
+gamb_cols <- paste0(c("E","L","P","Sm","Pm", "Im", "NonExistent"),"_gamb_count")
+
+Mos_sp_dist_sim <- cbind(species_simulation$timestep,
+                         rowSums(species_simulation[,arab_cols]),
+                         rowSums(species_simulation[,fun_cols]),
+                         rowSums(species_simulation[,gamb_cols]))
+
+species_plot <- function(Mos_pops){
+  plot(x = Mos_pops[,1], y = Mos_pops[,2], type = "l", col = cols[2],
+       ylim = c(0,140000), ylab = "Mosquito population size", xlab = "Days")
+  
+  grid()
   
-  subgraph clusterode {
-    style=filled;
-    color=lightblue;
-    node [shape = rectangle]
-    E
-    L
-    P
-    label='ODE'
-  }
-
-  node [shape = rectangle]
-  Sm
-  Pm
-  Im
-  NonExistent
+  sapply(3:4, function(x){
+    points(x = Mos_pops[,1], y = Mos_pops[,x], type = "l", col = cols[x])})
   
-  # edge definitions with the node IDs
-  E -> L -> P [color='green', label='1']
-  P -> Sm [color='green', label='2']
-  Sm -> Pm -> Im [color='orange', label='3']
-  Sm -> NonExistent [color='red', label='4']
-  Pm -> NonExistent [color='red', label='4']
-  Im -> NonExistent [color='red', label='4']
-  }")
+  legend("topright", legend = c("A. arab","A. fun","A. gamb"),
+         col = cols[-1], lty = 1, bty = "n", ncol = 1)
+}
+
+species_plot(Mos_sp_dist_sim)
+```
+
+### Seasonality
+
+The `malariasimulation` package has the capacity to simulate malaria transmission for a range of seasonal transmission profiles. This is achieved by specifying an annual rainfall profile that shapes mosquito population dynamics, thereby impacting malaria transmission.
```

Comment:
Add a note here that seasonality parameters can be generated using the umbrella package and link

Reply:
Added